rtsp.go

```diff
@@ -110,57 +112,67 @@ func (rc *rtspCamera) clientReconnectBackgroundWorker(codecInfo videoCodec) {
 				badState = true
 			} else {
 				res, err := rc.client.Options(rc.u)
-				if err != nil && (errors.Is(err, liberrors.ErrClientTerminated{}) ||
-					errors.Is(err, io.EOF) ||
-					errors.Is(err, syscall.EPIPE) ||
-					errors.Is(err, syscall.ECONNREFUSED)) {
-					rc.logger.Warnw("The rtsp client encountered an error, trying to reconnect", "url", rc.u, "error", err)
+				rc.logger.Debugf("Options reponse: %s, err: %s", res, err)
+				// Nick S:
+				// This error happens all the time on hardware we need to support & does not affect
+				// the performance of camera streaming. As a result, we ignore this error specifically
+				_, isErrClientInvalidState := err.(liberrors.ErrClientInvalidState)
+				if err != nil && !isErrClientInvalidState {
+					rc.logger.Warnf("The rtsp client encountered an error, trying to reconnect to %s, err: %s", rc.u, err)
```

Comment:
After upgrading to gortsplib v4 we were occasionally seeing `syscall.ECONNRESET` errors which did not trigger reconnection but did cause result in the camera no longer receiving RTP packets, which caused the connection to go dead & the camera to freeze.

As a result I changed this logic to consider all errors other than the specific error we see all the time as a condition that should trigger reconnection.